### PR TITLE
Google Sheets - new actions

### PR DESCRIPTION
**Pull request by michelle0927**

### CODE REVIEW

**Comment by compwright** [comment]
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/google_sheets/actions/find-row/find-row.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const sheets = this.googleSheets.sheets();
+
+    const colValues = (await sheets.spreadsheets.values.get({
+      spreadsheetId: this.sheetId,
+      range: `${this.sheetName}!${this.column}:${this.column}`,
+    })).data.values;
+    return colValues.map((value, index) => {
+      if (value == this.value)
+        return index + 1;
+    }).filter((value) => value);
```
Comment (compwright) [inline components/google_sheets/actions/find-row/find-row.js:58]: Would be better (easier to read and reason about) and more efficient to filter first, then map:

```suggestion
    return colValues.filter(value => value == this.value)
      .map((value, index) => index + 1);
```

File: components/google_sheets/actions/update-row/update-row.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const cells = this.cells;
+
+    // validate input
+    if (!cells || !cells.length) {
+      throw new Error("Please enter an array of elements in `Cells / Column Values`.");
+    } else if (!Array.isArray(cells)) {
+      throw new Error("Cell / Column data is not an array. Please enter an array of elements in `Cells / Column Values`.");
+    } else if (Array.isArray(cells[0])) {
+      throw new Error("Cell / Column data is a multi-dimensional array. A one-dimensional is expected.");
+    }
```
Comment (compwright) [inline components/google_sheets/actions/update-row/update-row.js:60]: For your consideration: [Else Considered Harmful](https://www.youtube.com/watch?v=XDqZemwVXUk)

```suggestion
    }
    if (!Array.isArray(cells)) {
      throw new Error("Cell / Column data is not an array. Please enter an array of elements in `Cells / Column Values`.");
    }
    if (Array.isArray(cells[0])) {
      throw new Error("Cell / Column data is a multi-dimensional array. A one-dimensional is expected.");
    }
```

**Review by compwright — CHANGES_REQUESTED** [review]
@michelle0927 great job on this! Found one typo that must be fixed before proceeding, and suggested some minor improvements in a few places.

File: components/google_sheets/actions/find-row/find-row.js (hunk 1 above)
Comment (michelle0927) [reply components/google_sheets/actions/find-row/find-row.js:58]: When I test this with filtering first, then mapping, it seems to actually change the index/row values. For instance, if I have a column C with value 'z' in rows 5 and 13, it should return [ 5, 13 ]. When I filter it first, I get [ 1, 2 ]. Is there a better way to write this where we can keep the correct row numbers?

File: components/google_sheets/actions/create-worksheet/create-worksheet.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      ],
+    },
+    title: {
+      type: "string",
+      label: "Title",
+      description: "The title of the new worksheet",
+    },
+  },
+  async run() {
+    const request = {
```
[the remark this answers is not in the document]
Comment (michelle0927) [reply components/google_sheets/actions/create-worksheet/create-worksheet.js:34]: I wasn't able to find a way to create a new worksheet without using batchUpdate. Would you be able to point me in the proper direction?

File: components/google_sheets/actions/find-row/find-row.js (hunk 1 above)
Comment (compwright) [reply components/google_sheets/actions/find-row/find-row.js:58]: Good point. In that case, you can combine both operations with reduce:

```suggestion
    return colValues.reduce((values, value, index) => {
      if (value == this.value) {
        values.push(index + 1);
      }
      return values;
    });
```

File: components/google_sheets/actions/create-worksheet/create-worksheet.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (compwright) [reply components/google_sheets/actions/create-worksheet/create-worksheet.js:34]: That's weird. Evidently there is not a way!

**Comment by compwright** [comment]
@michelle0927 looks like you have conflicts, be sure to resolve those too.

File: components/google_sheets/actions/find-row/find-row.js (hunk 1 above)
Comment (michelle0927) [reply components/google_sheets/actions/find-row/find-row.js]: Updated

File: components/google_sheets/actions/update-row/update-row.js (hunk 2 above)
Comment (michelle0927) [reply components/google_sheets/actions/update-row/update-row.js]: Updated

**Review by compwright — APPROVED** [review]

**Pull request merged**